sage/gen_split_lambda_constants.sage

```diff
@@ -0,0 +1,111 @@
+""" Generates the constants used in secp256k1_scalar_split_lambda.
+
+See the comments for secp256k1_scalar_split_lambda in src/scalar_impl.h for detailed explanations.
+"""
+
+load("secp256k1_params.sage")
+
+def inf_norm(v):
+    """Returns the infinity norm of a vector."""
+    return max(map(abs, v))
+
+def gauss_reduction(i1, i2):
+    v1, v2 = i1.copy(), i2.copy()
+    while True:
+        if inf_norm(v2) < inf_norm(v1):
+            v1, v2 = v2, v1
+        m = round( (v1[0]*v2[0] + v1[1]*v2[1]) // inf_norm(v1)**2 )
```

Comment:
No need to round anymore with `//`.

Reply:
Oh, I thought `/` vs `//` is just a python2 compatibility thing but I think what you had in mind is to avoid floating-point arithmetic. But this changed the semantics.

The correct thing is to `round` (to the nearest int) and not `floor` (for a reason), so that's why I used `/` and `round` instead of `//`. (It just happens that the output is the same *for our specific input*.)
   
So I now changed this to a version which
 * rounds to the nearest integer (what the textbooks do TM [1]),
 * relies integer arithmetic instead of floating-point arithmetic, and
 * should work in Python 2 and Python 3 because it does not use `/`.


[1] https://en.wikipedia.org/wiki/Lattice_reduction#In_two_dimensions Ok not a book but you get the point.

Reply:
Just saying, you could delete the whole thing and call `LLL` where all this has been worked out.